### PR TITLE
603 Use Multi-GPU in SlidingWindow example

### LINKED ISSUE #603: examples of multi-gpu sliding-window inference
**Is your feature request related to a problem? Please describe.**
would be great to update one of the segmentation inference examples to show the multi-gpu utilities

**Additional context**
one of the sliding window example
https://github.com/Project-MONAI/MONAI/blob/master/examples/segmentation_3d/unet_evaluation_dict.py

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #603 .

### Description
This PR shows how to use MONAI utility to get multi gpus and use them for SlidingWindow inference in the 3D segmentation example.
During testing, 2 GPUs worked as expected:
image

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: examples/segmentation_3d/unet_evaluation_dict.py [hunk 1]
```diff
@@ -62,24 +63,30 @@ def main():
...
+        model = torch.nn.DataParallel(model, device_ids=devices)
+
     model.eval()
     with torch.no_grad():
         metric_sum = 0.0
         metric_count = 0
         saver = NiftiSaver(output_dir="./output")
         for val_data in val_loader:
-            val_images, val_labels = val_data["img"].to(device), val_data["seg"].to(device)
+            val_images, val_labels = val_data["img"].to(devices[0]), val_data["seg"].to(devices[0])
```
Comment (wyli) [inline examples/segmentation_3d/unet_evaluation_dict.py:89]: will `.to(devices[0], non_blocking=True)` make it faster?

Comment (Nic-Ma) [reply examples/segmentation_3d/unet_evaluation_dict.py:89]: Hi @wyli ,

Thanks for your review.
According to https://discuss.pytorch.org/t/should-we-set-non-blocking-to-true/38234/4, I think here we have logic to save output into files, so `non_blocking=True` will not benefit performance.
And to keep this example straight-forward to show just multi-GPU inference, I suggest using the default value.
What do you think?

Thanks.

**Comment by Nic-Ma** [comment]
> have you tested the overall inference speed?

This example is very fast, I didn't test the inference speed. Do you think it's necessary?
Thanks.

**Comment by wyli** [comment]
> > have you tested the overall inference speed?
> 
> This example is very fast, I didn't test the inference speed. Do you think it's necessary?
> Thanks.

yes I think so, if it's slower than the previous setting, doesn't make sense to merge this PR

**Comment by Nic-Ma** [comment]
Hi @wyli ,

1. Single GPU time: 0.98s, Multi-GPU time: 3.51s, Multi-GPU with non_blocking time: 3.59s.
2. As you can see, my GPUs are very powerful, every GPU is only 36% used for this small example, so it makes sense that Multi-GPU is slower than single GPU here, when I increased the window size and batch size, the time consuming is almost the same.
If the image is very big, we can use bigger sliding window size and batch size in Multi-GPU mode.
3. I think you raised this task to show that we can support Multi-GPU for sliding window, isn't it?

Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

I tried to use more training data, a bigger window(512, 512, 512), and bigger batch size(8) for every GPU. the IO didn't affect much and there is no difference now:
Single GPU: 31.80s, Multi-GPU 31.89s.

Thanks.

**Pull request merged**